src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * Properties of the recurring `Job`.
+   */
+  public readonly jobProperties: JobProps;
+
+  /**
+   * Schedule for which to run the recurring job. By default, the job would be scheduled to run every minute.
+   */
+  public readonly schedule?: CronOptions;
+
+  /**
+   * Timezone in which the recurring job is scheduled.
+   */
+  public readonly timeZone?: string;
+
+  /**
+   * Concurrency policy for the recurring job. This specifies how the job scheduling would be handled based on which policy is selected.
+   */
+  public readonly concurrencyPolicy?: string;
+
+  /**
+   * Deadline to start next instance of job after the expected scheduling time. The job is considered as failed if it misses this deadline.
+   */
+  public readonly startingDeadline?: Duration;
+
+  /**
+   * Flag that tells to suspend the upcoming executions of the job.
+   */
+  public readonly suspend?: boolean;
+
+  /**
+   * The number of successful jobs to retain.
+   */
+  public readonly successfulJobsRetained?: number;
+
+  /**
+   * The number of failed jobs to retain.
+   */
+  public readonly failedJobsRetained?: number;
+
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  /**
+   * Represents the resource type.
+   */
+  public readonly resourceType = 'cronjob';
+
+  constructor(scope: Construct, id: string, props: CronJobProps) {
+    super(scope, id, {
+      restartPolicy: RestartPolicy.ON_FAILURE,
+      select: false, //Why is this false in Job?
+      ...props,
+    });
+
+    this.apiObject = new k8s.KubeCronJob(this, 'Resource', {
+      metadata: props.metadata,
+      spec: Lazy.any({ produce: () => this._toKube() }),
+    });
+
+    if (!this.isValidTimezone(props.timeZone)) {
+      throw new Error(`Invalid timezone: ${props.timeZone}`);
+    }
+
+    this.jobProperties = props.jobProperties;
+    this.schedule = props.schedule;
+    this.timeZone = props.timeZone;
+    this.concurrencyPolicy = props.concurrencyPolicy;
+    this.startingDeadline = props.startingDeadline;
+    this.suspend = props.suspend;
+    this.successfulJobsRetained = props.successfulJobsRetained;
+    this.failedJobsRetained = props.failedJobsRetained;
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.CronJobSpec {
+    return {
+      concurrencyPolicy: this.concurrencyPolicy ?? ConcurrencyPolicy.FORBID,
+      failedJobsHistoryLimit: this.failedJobsRetained,
+      jobTemplate: {
+        metadata: this.jobProperties.metadata,
+        spec: this.getJobSpec(),
+      },
+      schedule: this.getCronExpression(this.schedule),
+      startingDeadlineSeconds: this.startingDeadline?.toSeconds(),
+      successfulJobsHistoryLimit: this.successfulJobsRetained,
+      suspend: this.suspend,
+      timeZone: this.timeZone,
+    };
+  }
+
+  /**
+   * Returns the job spec.
+   */
+  private getJobSpec(): k8s.JobSpec {
```

Comment:
We usually refrain from using `get` for anything other than a getter (which is just direct property access in TS).
Also, start these conversion methods with `_`.  

Just some conventions we gathered along the way :)

```suggestion
  private _toJobSpec(): k8s.JobSpec {
```

Reply:
Will update it, thank you for letting me know about these conventions :)